packages/@react-aria/dnd/src/DragManager.ts

```diff
@@ -188,7 +208,23 @@ class DragSession {
     );
     this.updateValidDropTargets();
 
-    announce(this.formatMessage(MESSAGES[getDragModality()]));
+    let dragModality = getDragModality();
+    announce(this.formatMessage(MESSAGES[dragModality]));
+
+    if (dragModality === 'virtual') {
+      this.genericAncestor =
```

Comment:
Does adding `role="application"` to `document.body` work? Since the whole page gets placed into drag and drop mode, I think that would be ok?

Reply:
Hmm interesting. Then maybe the closest ancestor to the body? Might be faster to walk up the tree rather than querying all the elements in the whole document?

Reply:
I can give that a try. I actually tried refactoring to find the closest generic ancestor containing the dragTarget and all the dropTargets, but realized I would have to keep updating if a dropTarget were added outside of that generic ancestor.